File: task-03/C++/Sub-Task2/code2.cpp
#include <iostream>
#include <fstream>
#include <string>

int main() 
{
    // File streams
    std::ifstream infile("input.txt"); // Open input file in read mode
    std::ofstream outfile("output.txt"); // Open output file in write mode

    // Check if files are open
    if (!infile.is_open()) 
    {
        std::cerr << "Error" << std::endl;
        return 1;
    }
    if (!outfile.is_open()) 
    {
        std::cerr << "Error" << std::endl;
        return 1;
    }

    // Read from input.txt and write to output.txt
    std::string line;
    while (std::getline(infile, line)) 
    {
        outfile << line << std::endl;
    }

    // Close the files
    infile.close();
    outfile.close();

    std::cout << "Content copied from input.txt to output.txt" << std::endl;

    return 0;
}

File: task-03/java/Sub-Task2/code2.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.FileReader;
import java.io.FileWriter;
import java.io.IOException;

public class FileCopy 
{
    public static void main(String[] args) 
    {
        String inputFile = "input.txt";
        String outputFile = "output.txt";

        // Use try-with-resources to ensure resources are closed properly
        try (
            BufferedReader reader = new BufferedReader(new FileReader(inputFile));
            BufferedWriter writer = new BufferedWriter(new FileWriter(outputFile))
        ) {
            String line;
            // Read from input file and write to output file
            while ((line = reader.readLine()) != null) {
                writer.write(line);
                writer.newLine(); // Write a newline character
            }
            System.out.println("Content copied from " + inputFile + " to " + outputFile);
        } catch (IOException e) {
            System.err.println("An error occurred: " + e.getMessage());
        }
    }
}

